src/Groups/matrices/MatGrp.jl

```diff
@@ -158,13 +158,40 @@ function assign_from_description(G::MatrixGroup)
    elseif G.descr==:Sp G.X=GAP.Globals.Sp(G.deg,G.mat_iso.fr.codomain)
    elseif G.descr==Symbol("GO+") G.X=GAP.Globals.GO(1,G.deg,G.mat_iso.fr.codomain)
    elseif G.descr==Symbol("SO+") G.X=GAP.Globals.SO(1,G.deg,G.mat_iso.fr.codomain)
-   elseif G.descr==Symbol("Omega+") G.X=GAP.Globals.Omega(1,G.deg,Int(order(G.ring)))
+   elseif G.descr==Symbol("Omega+")
+      if G.deg==4 && order(G.ring)==2  # this is the only case SO(n,q) has more than one subgroup of index 2
+         L = GAP.Globals.SubgroupsOfIndexTwo(GAP.Globals.SO(1,G.deg,G.mat_iso.fr.codomain))
+         for y in L
+            _ranks = [GAP.Globals.Rank(u) for u in GAP.Globals.GeneratorsOfGroup(y)]
+            if all(r->iseven(r),_ranks)
+               G.X=y
+               break
+            end
+         end
+      else
+         G.X=GAP.Globals.SubgroupsOfIndexTwo(GAP.Globals.SO(1,G.deg,G.mat_iso.fr.codomain))[1]
+      end
    elseif G.descr==Symbol("GO-") G.X=GAP.Globals.GO(-1,G.deg,G.mat_iso.fr.codomain)
    elseif G.descr==Symbol("SO-") G.X=GAP.Globals.SO(-1,G.deg,G.mat_iso.fr.codomain)
-   elseif G.descr==Symbol("Omega-") G.X=GAP.Globals.Omega(-1,G.deg,Int(order(G.ring)))
-   elseif G.descr==:GO G.X=GAP.Globals.GO(0,G.deg,G.mat_iso.fr.codomain)
-   elseif G.descr==:SO G.X=GAP.Globals.SO(0,G.deg,G.mat_iso.fr.codomain)
-   elseif G.descr==:Omega G.X=GAP.Globals.Omega(0,G.deg,Int(order(G.ring)))
+   elseif G.descr==Symbol("Omega-") G.X=GAP.Globals.SubgroupsOfIndexTwo(GAP.Globals.SO(-1,G.deg,G.mat_iso.fr.codomain))[1]
+   elseif G.descr==:GO
+      if G.deg==1
+         G.X=GAP.Globals.Group(GapObj([GapObj([-GAP.Globals.One(G.mat_iso.f.riso.codomain)])]))
+      else
+         G.X=GAP.Globals.GO(0,G.deg,G.mat_iso.fr.codomain)
+      end
+   elseif G.descr==:SO
+      if G.deg==1
+         G.X=GAP.Globals.Group(GapObj([GapObj([GAP.Globals.One(G.mat_iso.f.riso.codomain)])]))
+      else
+         G.X=GAP.Globals.SO(0,G.deg,G.mat_iso.fr.codomain)
+      end
+   elseif G.descr==:Omega
+      if G.deg==1
+         G.X=GAP.Globals.Group(GapObj([GapObj([GAP.Globals.One(G.mat_iso.f.riso.codomain)])]))
+      else
+         G.X=GAP.Globals.SubgroupsOfIndexTwo(GAP.Globals.SO(0,G.deg,G.mat_iso.fr.codomain))[1]
```

Comment:
There should also be a FIXME/TODO comment here which can however simply point to the discussion above